lesson_10/libraries/src/loaders/xavier_cruz_loader.ts

```diff
@@ -0,0 +1,71 @@
+import csv from 'csv-parser';
+import fs from 'fs';
+import { Credit, MediaItem, Role } from '../models/index.js';
+import { Loader } from './loader.js';
+
+export class XavierCruzLoader implements Loader {
+  getLoaderName(): string {
+    return 'xaviercruz';
+  }
+
+  async loadData(): Promise<MediaItem[]> {
+    const credits = await this.loadCredits();
+    const mediaItems = await this.loadMediaItems();
+
+    console.log(
+      `Loaded ${credits.length} credits and ${mediaItems.length} media items`,
+    );
+
+    let counter = 0;
+    for (let i = 0; i < mediaItems.length; i++) {
+      if (i < mediaItems.length / 2) {
+        mediaItems[i].addCredit(credits[counter] as Credit);
+        counter++;
+        mediaItems[i].addCredit(credits[counter] as Credit);
```

Comment:
This is going to fail if the credits are not sorted by media item id. I'll make this more explicit by updating the example csv file.

Reply:
I agree it was a quick 'fix' will fix since some time freed up!